### PR TITLE
WNCM-57 update in django simple history integration and tests

### PR DESCRIPTION
**Pull request by goyal1092**
## Description
Updated The way we are fetching changed fields
Previously we stored changed fields to change_reason field for 2 reason:
1) Avoid querying history diff for the delta between prev obj state
2) Some forms are heavily customized so we were not able to get all detail of the changes

To fix currently faced issues:
* Use django simple history diff to get changed fields
* In case of dataset or collection upload from task - update change reason from task or admin so we know:
       *  If dataset is uploaded is there is no change reason given
       *  If collection is uploaded is there is no change reason given
       *  If group is updated by task
       * If dataset is updated by task

Updated tests accordingly

## Related Issue
https://wazimap.atlassian.net/browse/WNCM-57
## Checklist

- [ ]  🚀 is the code ready to be merged and go live?
- [ ]  🛠 does it work (build) locally

### Pull Request

- [ ]  📰 good title
- [ ]  📝good description
- [ ]  🔖 issue linked
- [ ]  📖 changelog filled out

### Commits

- [ ]  commits are clean
- [ ]  commit messages are clean

### Code Quality

- [ ]  🚧 no commented out code
- [ ]  🖨 no unnecessary logging
- [ ]  🎱 no magic numbers
- [ ]  black was run locally (as part of the pre-commit hook)

### Testing

- [ ]  ✅ added (appropriate) unit tests
- [ ]  💢 edge cases in tests were considered
- [ ]  ✅ ran tests locally & are passing

### CODE REVIEW

**Review by michaelglenister — COMMENTED** [review]
Changes are good, change reason and the the revert feature work properly. Also good test coverage

**Review by michaelglenister — APPROVED** [review]

**Comment by jbothma** [comment]
Either there are no tests for the changed fields functionality, or the tests aren't working, because changed fields aren't being shown here

Screenshot_2022-01-20_16-25-37

File: wazimap_ng/datasets/management/commands/port_admin_history.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            instance_model = ct.model_class()
+
+            # In case we have moved apps from model
+            # we moved around bunch of models on intial stage
+            if not instance_model:
+                continue
+            try:
+                # Get instance
+                instance = instance_model.objects.get(id=log_entry.object_id)
+            except (instance_model.DoesNotExist, FieldError):
```
Comment (michaelglenister) [inline wazimap_ng/datasets/management/commands/port_admin_history.py:31]: Is instance_model.DoesNotExist needed if we are checking it above?

**Review by michaelglenister — COMMENTED** [review]
Added one question, otherwise happy for this to move forwards as is

File: wazimap_ng/datasets/management/commands/port_admin_history.py (hunk 1 above)
Comment (goyal1092) [reply wazimap_ng/datasets/management/commands/port_admin_history.py:31]: @michaelglenister 
yup... if there is history for deleted objects. I am certain that the Django admin log also holds info about deleted data so it is just to make sure that the script does not fail.
We also changed apps for some models. so it's just to make sure we have the object present.

FieldError is where id does not exist. One case I was able to find was Token Model
Currently, we have simple history applied only on our main models and all of them have an id

**Pull request merged**